src/evict/evict_lru.c

```diff
@@ -909,20 +940,20 @@ __evict_lru_walk(WT_SESSION_IMPL *session)
 		}
 	}
 
-	queue->evict_current = queue->evict_queue;
-	__wt_spin_unlock(session, &queue->evict_lock);
-
-	/*
-	 * Now we can set the next queue.
-	 */
-	__wt_spin_lock(session, &cache->evict_queue_lock);
-	if (cache->evict_current_queue->evict_current == NULL)
+	if (__evict_queue_empty(queue)) {
+		/*
+		 * This score varies between 0 (if the queue hasn't been empty
+		 * for a long time) and 100 (if the queue has been empty the
+		 * last 10 times we filled up.
+		 */
+		cache->evict_empty_score = WT_MIN(100,
+		    cache->evict_empty_score + WT_EVICT_EMPTY_SCORE_BUMP);
```

Comment:
1. I've moved the comment into `cache.h`.
2. I disagree about the condition -- we have figured out which queue to fill above, in the ommon case we don't choose the current queue the net is the same as previously.  If we are evicting so fast that we start filling the current queue, we want to know whether it was drained and use that for the empty score.
3. I've moved this code earlier.
4. I've changed the code to just mark the queue being filled empty after locking so that subsequent requests for a page will switch to the other queue.